### DISCUSSION TITLE
Why is strict mode required

### DISCUSSION THREAD

**Discussion by efreethy** [colinhacks/zod#1755] [Q&A] [answered]
Our projects are a long way away from being able to flip on strict mode. Zod requires strict mode. Is there any compromise? Perhaps some minimum set of options that we can turn on that zod needs.

**Comment by santosmarco-caribou** [comment]
Mainly because of `strictNullChecks`. Can you at least turn this one on?

From the TS docs:

> When `strictNullChecks` is `false`, `null` and `undefined` are effectively ignored by the language. This can lead to unexpected errors at runtime.

This will cause the inference that Zod can provide to be probably wrong sometimes.

The library will work just as normal, but you won't be able to trust the inference.

**Comment by colinhacks** [comment, marked as answer]
You can use Zod without strict mode but the inferred types will be incorrect. Specifically, all object keys will be optional. There may be other unexpected behaviors but that's the main one. If that's okay for your team then you don't need strict mode.
